### PR TITLE
Fix query test.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

**Comment by ADmad** [comment]
So we get different failures on PHP 8.1 and PHP < 8.1. Someone more familiar with the `FunctionsBuilder` then me should look into this. 

PHP < 8.1
```
1) Cake\Test\TestCase\Database\QueryTest::testSQLFunctions
Failed asserting that two arrays are identical.
--- Expected
+++ Actual
@@ @@
     'de' => '18'
     'me' => '3'
     'ye' => '2007'
-    'wd' => 1
-    'dow' => 1
+    'wd' => '1'
+    'dow' => '1'
     'addDays' => '2007-03-20'
     'substractYears' => '2005-03-18'
 )

/home/runner/work/cakephp/cakephp/tests/TestCase/Database/QueryTest.php:3539
```

PHP 8.1
```
1) Cake\Test\TestCase\Database\QueryTest::testSQLFunctions
TypeError: ltrim(): Argument #1 ($string) must be of type string, int given

/home/runner/work/cakephp/cakephp/tests/TestCase/Database/QueryTest.php:3523
```

**Comment by othercorey** [comment]
This is from PDO changes in 8.1 where the driver return proper types.

https://github.com/php/php-src/blob/PHP-8.1/UPGRADING#L131

**Comment by ADmad** [comment]
So we should change the expected values based on PHP version in the tests?

**Comment by othercorey** [comment]
> So we should change the expected values based on PHP version in the tests?

Either that or make the test looser == instead of ===.

**Comment by ADmad** [comment]
> Either that or make the test looser == instead of ===.

I think it's better to make the difference stand out instead of hiding them.

File: tests/TestCase/Database/QueryTest.php [hunk 1]
```diff
@@ -3520,23 +3520,41 @@ function ($q) {
...
-            'ye' => '2007',
-            'wd' => '1', // Sunday
-            'dow' => '1',
-            'addDays' => '2007-03-20',
-            'substractYears' => '2005-03-18',
-        ];
-        $this->assertEquals($expected, $result[0]);
+
+        if (PHP_VERSION_ID < 80100) {
+            $result[0]['m'] = ltrim($result[0]['m'], '0');
```
Comment (othercorey) [inline tests/TestCase/Database/QueryTest.php:3528]: I wonder if this is still necessary. Maybe it is with mysql 5.6.

**Pull request merged**